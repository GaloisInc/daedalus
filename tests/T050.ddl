

def V = Choose {
  number = 1 as int;
  other  = Match "X";
}


def LookupInt (arr : [V]) i = Default 0 {
  @n = Index arr i;
  commit;
  n is number;
}

def arr : [V] =
  [ {| number = 1 |}, {| other = "Y" |} ]

def Main = {
  x1 = LookupInt arr 0;
  -- x11 = LookupInt arr 1; -- parse error
  x2 = LookupInt arr 2;
}

def Default x P = P <| ^ x
