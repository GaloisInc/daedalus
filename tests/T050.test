T050.ddl
--interp=T050.ddl
