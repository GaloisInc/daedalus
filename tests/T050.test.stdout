--- Found 1 results:
{ x1: 1
, x2: 0
}
